### DISCUSSION TITLE
Is there a reason the containerd worker runtime arch is limited to only 3 arch?

### DISCUSSION THREAD

**Discussion by rdrgmnzs** [concourse/concourse#8451] [Help & Support]
Is there a reason containerd is limited by to only use [3 arch here?](https://github.com/concourse/concourse/blob/8c8eb0565101b287107fe9d36ef970505ad166b7/worker/runtime/spec/seccomp.go#L7)  this limits the ability of someone running https://github.com/robinhuiser/concourse-arm64 in being able to run a worker on an  apple M1 laptop.

**Comment by rdrgmnzs** [comment]
Digging some more it looks like it was just copied from guardian in https://github.com/concourse/concourse/pull/5228 and what is in guardian was added as a "unprivilegedbundle" 6 years ago.

Anyone know of any reason not to change this to allow M1 machines to run?

**Comment by marco-m** [comment]
I think you are confusing architecture, OS and what Concourse supports for containerization/virtualization.
Containerd is for Linux only. On macOS, either amd64 or arm64, you cannot use containerd, you will still have to run the Concourse tasks directly on the host or VM, no containers.
If you build the Concourse worker for OS darwin and ARCH arm64, everything will work fine, we run this setup in production.
Or am I missing something?

**Comment by rdrgmnzs** [comment]
@marco-m I was referring to how if you use https://github.com/robinhuiser/concourse-arm64 you can build a arm64 version of the docker image 'concourse/concourse' and then run it with docker-compose in the same you would the [regular quick start guide](https://concourse-ci.org/quick-start.html). 

That all works until concourse tries to startup a worker inside the `concourse` container utilizing containerd, those workers are limited to only allow execution of binaries with the [3 arch here](https://github.com/concourse/concourse/blob/8c8eb0565101b287107fe9d36ef970505ad166b7/worker/runtime/spec/seccomp.go#L7) SecComp, at which point you get the error `start process: backend error: Exit status: 500, message: {"Type":"","Message":"proc start: OCI runtime exec failed: exec failed: container_linux.go:380: starting container process caused: exec format error: unknown","Handle":"","ProcessID":"","Binary":""}` as it's trying to execute arm64 binaries.

Sorry hopefully what I described makes more sense now.

**Comment by xtremerui** [comment]
@rdrgmnzs concourse/dev now supports arm64. If you run docker compose on a M1 machine, it should work. When I running it on my m1 laptop I dont see the error you mentioned. Could you try and see if it works for you?

**Comment by rdrgmnzs** [comment]
Hey @xtremerui unfortunately I no longer work at the same place and can't easily test this. Will close the discussion out for now.
